app/models/reception/resource_factory.rb

```diff
@@ -148,5 +144,37 @@ def duplicate_containers
       cons.fetch('Well', []).map { |well| [well.position, well.plate] }.uniq! ||
         cons.fetch('Tube', []).map(&:barcode).uniq! || cons.fetch('Plate', []).map(&:barcode).uniq!
     end
+
+    def library_type_cache
+      # Used to reduce database queries.
+      @library_type_cache ||= LibraryType.all.index_by(&:name)
+    end
+  end
+
+  # A helper that deals with the creation of library type specific records.
```

Comment:
Advisory. Not suggesting we change it in this release. You can use ActiveModel::Model here which is the recommended way.